cmake/FindWayland.cmake

```
@@ -0,0 +1,12 @@
+find_package(PkgConfig)
+
+if(PKG_CONFIG_FOUND)
+    pkg_check_modules(WAYLAND_CLIENT REQUIRED IMPORTED_TARGET wayland-client)
+
+    find_path(WAYLAND_CLIENT_INCLUDE_DIR NAMES wayland-client.h HINTS ${WAYLAND_CLIENT_INCLUDE_DIRS})
+    add_library(Wayland::client ALIAS PkgConfig::WAYLAND_CLIENT)
+
+    set(Wayland_INCLUDE_DIR ${WAYLAND_CLIENT_INCLUDE_DIR})
+    set(Wayland_LIBRARIES ${WAYLAND_CLIENT_LIBRARIES})
+    find_package_handle_standard_args(Wayland DEFAULT_MSG Wayland_LIBRARIES Wayland_INCLUDE_DIR)
+endif()
```

Using `REQUIRED` in the find module itself is considered bad practice; it should be specified when calling `find_package()` (and it is). Also, considering the changes made in the patch https://github.com/yuzu-emu/yuzu/pull/9395, I'd suggest changing the module as following:

```suggestion
if(PKG_CONFIG_FOUND)
    pkg_search_module(WAYLAND_CLIENT IMPORTED_TARGET wayland-client)
endif()

include(FindPackageHandleStandardArgs)
find_package_handle_standard_args(Wayland
    REQUIRED_VARS
        WAYLAND_CLIENT_LIBRARIES
        WAYLAND_CLIENT_INCLUDE_DIR
    VERSION_VAR WAYLAND_CLIENT_VERSION
)

if (Wayland_FOUND)
    add_library(Wayland::client ALIAS PkgConfig::WAYLAND_CLIENT)
endif()
```